papaya-web/src/main/java/pl/edu/agh/papaya/rest/projects/service/ProjectsRestService.java

```diff
@@ -0,0 +1,154 @@
+import pl.edu.agh.papaya.service.user.UserService;
+import pl.edu.agh.papaya.util.BadRequestException;
+import pl.edu.agh.papaya.util.ForbiddenAccessException;
+import pl.edu.agh.papaya.util.ResourceNotFoundException;
+
+@Service
+public class ProjectsRestService {
+
+    @Autowired
+    private UserContext userContext;
+
+    @Autowired
+    private UserService userService;
+
+    @Autowired
+    private UserIdentificationService userIdentificationService;
+
+    @Autowired
+    private ProjectService projectService;
+
+    @Autowired
+    private ProjectMapper projectMapper;
+
+    @Autowired
+    private UserRoleMapper userRoleMapper;
+
+    @Autowired
+    private UserInProjectMapper userInProjectMapper;
+
+    public ResponseEntity<ProjectDto> addProject(ProjectDto projectDto) {
+        Project created = projectService.newProject()
+                .withName(projectDto.getName())
+                .withDescription(projectDto.getDescription())
+                .withInitialCoefficient(projectDto.getInitialCoefficient())
+                .create();
+        return ResponseEntity.ok(projectMapper.mapToApi(created));
+    }
+
+    public ResponseEntity<ProjectMemberDto> addUserToProject(UserIdentificationDto userIdentification, Long projectId) {
+        User user = userIdentificationService.identify(userIdentification)
+                .orElseThrow(BadRequestException::new);
+
+        Project project = projectService.getProjectById(projectId)
+                .orElseThrow(ResourceNotFoundException::new);
+
+        try {
+            projectService.setUserRole(project, user, UserRole.MEMBER);
+        } catch (UserNotAuthorizedException e) {
+            throw new ForbiddenAccessException(e);
+        }
+
+        return ResponseEntity.ok(new ProjectMemberDto()
+                .userId(user.getId())
+                .role(UserRoleDto.MEMBER));
+    }
+
+    public ResponseEntity<ProjectDto> getProjectById(Long id) {
+        Project project = getValidProject(id);
+
+        return ResponseEntity.ok(projectMapper.mapToApi(project));
+    }
+
+    public Project getValidProject(Long id) {
+        Project project = projectService.getProjectById(id)
+                .orElseThrow(ResourceNotFoundException::new);
+
+        if (!projectService.isUserInProject(project, userContext.getUserId())) {
+            throw new ForbiddenAccessException();
+        }
+
+        return project;
+    }
+
+    public ResponseEntity<List<ProjectDto>> getProjects() {
+        List<Project> projects = projectService.getUserProjects(userContext.getUserId());
+        return ResponseEntity.ok(projectMapper.mapToApi(projects));
+    }
+
+    public ResponseEntity<List<ProjectMemberDto>> getUsersFromProject(Long projectId) {
+        Project project = projectService.getProjectById(projectId)
+                .orElseThrow(ResourceNotFoundException::new);
+
+        List<UserInProject> usersInProject = projectService.getUsersInProject(project);
```

Comment:
Nie wiem, czy bym już w `ProjectService` nie filtrowała `INACTIVE`?

Reply:
Słuszna uwaga, to dowodzi że obiekty bazodanowe nie powinny być tym samym co obiekty za pomocą których komunikuje się web z corem.